beacon/engine/types.go

```diff
@@ -270,7 +275,18 @@ func ExecutableDataToBlockNoHash(data ExecutableData, versionedHashes []common.H
 	// ExecutableData before withdrawals are enabled by marshaling
 	// Withdrawals as the json null value.
 	var withdrawalsRoot *common.Hash
-	if data.Withdrawals != nil {
+	if bType.HasOptimismWithdrawalsRoot(data.Timestamp) {
+		if data.WithdrawalsRoot == nil {
+			return nil, fmt.Errorf("attribute WithdrawalsRoot is required for Isthmus blocks")
```

Comment:
I appreciate we were trying to uncouple the feature "OptimismWithdrawalsRoot" from the fork ("Isthmus" in this case, but could be a different one). These error messages don't really achieve that decoupling.